src/DSharp.Script/src/System/Enum.js

```diff
@@ -46,4 +46,16 @@ Enum.hasFlag = function (enumValue, flag) {
 Enum.isDefined = function (enumeration, value) {
     var map = Enum.map[enumeration._name];
     return isValue(map.Values[value]) || isValue(map.Keys[value]);
-};
\ No newline at end of file
+};
+
+Enum.parse = function (enumeration, value, ignoreCase) {
+    var map = Enum.map[enumeration._name];
+    if (map && ignoreCase) {
+        for (var p in map.Keys) {
```

Comment:
Can be simplified using `.filter`
```js
var possibleValues = map.Keys.filter(p => p.toLowerCase() === value.toLowerCase());
if (possibleValues.length === 1) return map.Keys[possibleValues[0]];
return null;
```

Reply:
sadly Keys is not an array, so we would have to do `Object.GetOwnPropertyNames` to use filter